marshmallow >= 3.1.0 not working in Python<3.5.3

Hello!

I'm getting the following error when using `marshmallow` 3.1.0 or 3.1.1 with `pytest`:
```
============================================== test session starts ===============================================
platform linux -- Python 3.5.2, pytest-5.1.2, py-1.8.0, pluggy-0.13.0
rootdir: ***, inifile: setup.cfg
plugins: cov-2.7.1, asyncio-0.10.0
collected 31 items / 1 errors / 30 selected                                                                      

===================================================== ERRORS =====================================================
______________________________ ERROR collecting tests_async/test_event_executor.py _______________________________
*** imports inside my project ***
    from marshmallow import Schema, fields, decorators
../../.local/share/virtualenvs/gateway-WH935Ska/lib/python3.5/site-packages/marshmallow/__init__.py:1: in <module>
    from marshmallow.schema import Schema, SchemaOpts
../../.local/share/virtualenvs/gateway-WH935Ska/lib/python3.5/site-packages/marshmallow/schema.py:14: in <module>
    from marshmallow import base, fields as ma_fields, class_registry
../../.local/share/virtualenvs/gateway-WH935Ska/lib/python3.5/site-packages/marshmallow/fields.py:434: in <module>
    class Nested(Field):
../../.local/share/virtualenvs/gateway-WH935Ska/lib/python3.5/site-packages/marshmallow/fields.py:480: in Nested
    unknown: str = None,
../../miniconda3/envs/gateway3.5.2/lib/python3.5/typing.py:552: in __getitem__
    dict(self.__dict__), parameters, _root=True)
../../miniconda3/envs/gateway3.5.2/lib/python3.5/typing.py:512: in __new__
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
../../miniconda3/envs/gateway3.5.2/lib/python3.5/typing.py:512: in <genexpr>
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
../../miniconda3/envs/gateway3.5.2/lib/python3.5/typing.py:1077: in __subclasscheck__
    if super().__subclasscheck__(cls):
../../.local/share/virtualenvs/gateway-WH935Ska/lib/python3.5/abc.py:225: in __subclasscheck__
    for scls in cls.__subclasses__():
E   TypeError: descriptor '__subclasses__' of 'type' object needs an argument

```
The last working version was 3.0.5.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -8,6 +8,10 @@ Features:
 
 - Add type annotations to ``marshmallow.schema`` and ``marshmallow.validate`` (:pr:`1407`, :issue:`663`).
 
+Bug fixes:
+
+- Fix compatibility with Python < 3.5.3 (:issue:`1409`). Thanks :user:`lukaszdudek-silvair` for reporting.
+
 Refactoring:
 
 - Remove unnecessary ``BaseSchema`` superclass (:pr:`1406`).

--- a/src/marshmallow/fields.py
+++ b/src/marshmallow/fields.py
@@ -471,7 +471,7 @@ class Nested(Field):
 
     def __init__(
         self,
-        nested: typing.Union[SchemaABC, typing.Type[SchemaABC], str],
+        nested: typing.Union[SchemaABC, type, str],
         *,
         default: typing.Any = missing_,
         only: types.StrSequenceOrSet = None,
@@ -664,9 +664,7 @@ class List(Field):
 
     default_error_messages = {"invalid": "Not a valid list."}
 
-    def __init__(
-        self, cls_or_instance: typing.Union[Field, typing.Type[Field]], **kwargs
-    ):
+    def __init__(self, cls_or_instance: typing.Union[Field, type], **kwargs):
         super().__init__(**kwargs)
         try:
             self.inner = resolve_field_instance(cls_or_instance)
@@ -1427,8 +1425,8 @@ class Mapping(Field):
 
     def __init__(
         self,
-        keys: typing.Union[Field, typing.Type[Field]] = None,
-        values: typing.Union[Field, typing.Type[Field]] = None,
+        keys: typing.Union[Field, type] = None,
+        values: typing.Union[Field, type] = None,
         **kwargs
     ):
         super().__init__(**kwargs)

--- a/src/marshmallow/schema.py
+++ b/src/marshmallow/schema.py
@@ -415,10 +415,10 @@ class Schema(base.SchemaABC, metaclass=SchemaMeta):
     @classmethod
     def from_dict(
         cls,
-        fields: typing.Dict[str, typing.Union[ma_fields.Field, typing.Type]],
+        fields: typing.Dict[str, typing.Union[ma_fields.Field, type]],
         *,
         name: str = "GeneratedSchema"
-    ) -> typing.Type["Schema"]:
+    ) -> type:
         """Generate a `Schema` class given a dictionary of fields.
 
         .. code-block:: python

